packages/gatsby-image/src/index.js

```diff
@@ -69,6 +76,95 @@ function getIO() {
   return io
 }
 
+function generateImageSources(imageVariants) {
+  const initialImage = imageVariants[0]
+
+  if (imageVariants.length === 1) {
+    return initialImage.srcSetWebp ? (
+      <source
+        type={`image/webp`}
+        media={initialImage.media}
+        srcSet={initialImage.srcSetWebp}
+        sizes={initialImage.sizes}
+      />
+    ) : null
+  }
+
+  return imageVariants.map(variant => {
+    const webPSource = variant.srcSetWebp ? (
+      <source
+        key={`${variant.src}-webp`}
+        type="image/webp"
+        media={variant.media}
+        srcSet={variant.srcSetWebp}
+        sizes={variant.sizes}
+      />
+    ) : (
+      undefined
+    )
+
+    const nonWebPSource = (
+      <source
```

Comment:
When converting gatsby-image to use picture tag originally - I reasoned that the last source was the 'most compatible' and very likely the same image format that would be used in the fallback img tag as well.   Seems reasonable to add I suppose - just keeping in mind that it isn't always Jpeg.

Reply:
> Seems reasonable to add I suppose - just keeping in mind that it isn't always Jpeg.

The mimetype can probably be added to the metadata object during build time, so that it's just another property that gets used if available? There is an issue with that and sharp code atm, where the mozjpeg support may not process a png input, but output a file with jpg extension that's still png data, without raising an error or warning.

There is a mimetype crate that I linked to on the relevant issue about it that could help avoid that.